src/game/logic/object/weapon.cpp

```diff
@@ -1195,15 +1370,236 @@ bool Weapon::Compute_Approach_Target(const Object *source_obj,
+                vector.z = vector.z / length;
+
+                if (angle_offset != 0.0f) {
+                    float angle = GameMath::Atan2(vector.y, vector.x);
+                    vector.x = GameMath::Cos(angle + angle_offset);
+                    vector.y = GameMath::Sin(angle + angle_offset);
+                }
+
+                float range = Get_Attack_Range(source_obj) * 0.9f;
+                approach_target_pos.x = range * vector.x + pos->x;
+                approach_target_pos.y = range * vector.y + pos->y;
+                approach_target_pos.z = range * vector.z + pos->z;
+
+                if (source_obj->Get_AI_Update_Interface() != nullptr) {
+                    const AIUpdateInterface *update = source_obj->Get_AI_Update_Interface();
+
+                    if (update->Is_Aircraft_That_Adjusts_Destination()) {
+                        g_theAI->Get_Pathfinder()->Adjust_Target_Destination(
+                            source_obj, target_obj, target_pos, this, &approach_target_pos);
+                    }
+                }
+
+                return false;
+            }
+        } else {
+            approach_target_pos = *source_obj->Get_Position();
+            return true;
+        }
+    } else {
+        captainslog_dbgassert(Get_Attack_Range(source_obj) * 0.9f > min_range, "Min attack range is too near attack range.");
+        const Coord3D *pos2 = source_obj->Get_Position();
+        Coord3D vector;
+        vector.x = pos2->x - pos->x;
+        vector.x = pos2->y - pos->y;
+        vector.z = 0.0f;
+        vector.Normalize();
+
+        if (source_obj->Is_Above_Terrain()) {
+            float angle = source_obj->Get_Orientation() - GameMath::Atan2(-vector.y, -vector.x);
+
+            if (angle > DEG_TO_RADF(360.0f)) {
+                angle = angle - DEG_TO_RADF(360.0f);
+            }
+
+            if (angle < DEG_TO_RADF(-360.0f)) {
+                angle = angle + DEG_TO_RADF(360.0f);
+            }
+
+            if (GameMath::Fabs(angle) < DEG_TO_RADF(90.0f)) {
+                vector.x = -vector.x;
+                vector.y = -vector.y;
+                vector.z = -vector.z;
+            }
+        }
+
+        if (angle_offset != 0.0f) {
+            float angle = GameMath::Atan2(vector.y, vector.x);
+            vector.x = GameMath::Cos(angle + angle_offset);
+            vector.y = GameMath::Sin(angle + angle_offset);
+        }
+
+        float range2 = (Get_Attack_Range(source_obj) - min_range) / 2.0f;
+
+        if (target_obj != nullptr) {
+            range2 += target_obj->Get_Geometry_Info().Get_Bounding_Circle_Radius();
+        }
+
+        range2 += source_obj->Get_Geometry_Info().Get_Bounding_Circle_Radius();
+        approach_target_pos.x = range2 * vector.x + pos->x;
+        approach_target_pos.y = range2 * vector.y + pos->y;
+        approach_target_pos.z = range2 * vector.z + pos->z;
+        Clip_To_Terrain_Extent(approach_target_pos);
+        return false;
```

Comment:
`return false` is correct? Would have expected `true`, because `approach_target_pos` is set.

Reply:
Yes this is correct.